src/main/java/net/pms/util/InfoDb.java

```diff
@@ -36,18 +37,18 @@ public InfoDb() {
 
 
 	private void askAndInsert(File f, String formattedName) {
-		synchronized (db) {
```

Comment:
It was this commit that starts my hard drive going crazy, probably the removal of this line since that was preventing us from parsing all files at once, which makes me think maybe it's the hash computation method that is causing that HDD strain... Looking into it more now

Reply:
Yes it seems like that's the case. I replaced the contents of `getHash` with `return null` and it all works nicely now. Maybe there is a more efficient way to compute hashes

Reply:
@SubJunk The hash computation is defined by OpenSubtitles. I haven't changed it, and I don't think there's much to be gained there.

I still don't quite understand what this changed though, but the commit was so long ago that I'll have to look at it again to see what I really did.

Reply:
Looking at the docs http://trac.opensubtitles.org/projects/opensubtitles/wiki/HashSourceCodes#Java it seems like we are using a different method to them for calculating the hash - we are using the newer `path` syntax but it's ultimately still using the old `file` syntax because of the way `findSubtitles` converts the `RealFile` to `Path` via `File` instead of really using `Path`. So we could go down the route of actually implementing `Path` support or we could try using the unmodified code from `OpenSubtitles` since it may be faster for non-streaming files.

It looks like our current code turns everything into a stream and then loops through the file as fast as it can to get to the last chunk, whereas the code example on `OpenSubtitles` skips to the end which might end up faster.

Reply:
@SubJunk I don't think the ```Path```/```File``` difference matters much, but looping through a file of several GB versus skipping probably makes a huge difference. I didn't really look at that code, I assumes it was just copy/paste from OpenSubtitles. I might have been too, but maybe they have since found a better algorithm. Updating is absolutely seems like a good idea.

Reply:
Weirdly enough it doesn't seem to make much difference, it's taking about 30ms per file anyway, so it looks like this wasn't the cause of the problem